e2e-tests/testdata/echo_client/ts/client.ts

```diff
@@ -815,7 +818,7 @@ class BaseClient {
             if (typeof auth === "function") {
                 this.authGenerator = auth
```

Comment:
What if the function doesn't return a promise; wouldn't the `await this.authGenerator()` fail then?

Reply:
yeah, i was thinking the same and then tested, and `await` on a value that is not a promise, returns the value itself in JS.

do you know if there are integrations tests / some other tests for the generated code somewhere in the repo where this behaviour gets tested?

Reply:
I had another look at this one, and there might be some performance implications with the added `await`, i'll add a check around this. thanks!